From f8e9b5426ec5ba3497c20b190d74f78b9ccdc0f2 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Fri, 24 Jan 2025 10:49:31 +0200
Subject: [PATCH] [SW-227776] xdpp/pe: discard invalid temp sensor value

It is observed that sometimes the temperature value
read from xdpp and xdpe sensor is 0xFFFF, Hence as a
WA his patch adds support to ignore these values.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I7c0ae132bbecb3ba79be6c1189d4d7f59e18ea65
---

diff --git a/drivers/sensor/xdpe132g5h/xdpe132g5h.c b/drivers/sensor/xdpe132g5h/xdpe132g5h.c
index a8ae2ee..9c0018b 100644
--- a/drivers/sensor/xdpe132g5h/xdpe132g5h.c
+++ b/drivers/sensor/xdpe132g5h/xdpe132g5h.c
@@ -35,6 +35,7 @@
 #define XDPE132G5H_TEMP_WRN_INTR_PRIO	0x0
 #define XDPE132G5H_VOUT_EXPONENT	0xC
 #define XDPE1A2G5B_VOUT_EXPONENT	0xA
+#define XDPE132G5H_CTEMP_MAX_VALID	130
 
 #define SMB_COUNT			2
 #define MFR_MODEL_LB			0x7C
@@ -395,7 +396,11 @@
 			return -EIO;
 		}
 
-		drv_data->vdd_temp_data.temp = val;
+		if (val > XDPE132G5H_CTEMP_MAX_VALID) {
+			LOG_DBG("Invalid sensor value ignoring!");
+		} else {
+			drv_data->vdd_temp_data.temp = val;
+		}
 		break;
 	case XDPE132G5H_SENSOR_CHAN_LOOP_1_VDD_TEMP_PEAK:
 		ret = pmbus_read_data_word(&drv_data->client,
diff --git a/drivers/sensor/xdpp1100/xdpp1100.c b/drivers/sensor/xdpp1100/xdpp1100.c
index 431f2a6..a077dd3 100644
--- a/drivers/sensor/xdpp1100/xdpp1100.c
+++ b/drivers/sensor/xdpp1100/xdpp1100.c
@@ -31,6 +31,7 @@
 #define PMBUS_USER_DATA_00		0xB0
 
 #define XDPP1100_TEMP_WRN_INTR_PRIO	0x0
+#define XDPP1100_CTEMP_MAX_VALID	130
 
 #define SMB_COUNT			2
 #define USERDATA_00_BYTES		16
@@ -203,7 +204,12 @@
 				ret = -EIO;
 				goto error;
 			}
-			*(data_map->val_addr) = val;
+			if (ch == XDPP1100_SENSOR_CHAN_LOOP_1_VDD_TEMP &&
+				val > XDPP1100_CTEMP_MAX_VALID) {
+				LOG_DBG("Invalid sensor value ignoring!");
+			} else {
+				*(data_map->val_addr) = val;
+			}
 		}
 	} else {
 		ret = -ENOTSUP;
